src/BenchmarksClient/Startup.cs

```diff
@@ -100,72 +100,88 @@ public static int Main(string[] args)
+                                worker = WorkerFactory.CreateWorker(job);
+                            }
 
                             if (worker == null)
                             {
                                 Log($"Error while creating the worker");
-                                job.State = ClientState.Deleting;
+                                job.State = ClientJobState.Deleting;
                             }
                             else
                             {
-                                await worker.StartAsync();
+                                await worker.StartJobAsync(job);
                             }
                         }
                         catch (Exception e)
                         {
                             Log($"An unexpected error occured while starting the job {job.Id}");
                             Log(e.ToString());
 
-                            job.State = ClientState.Deleting;
+                            job.State = ClientJobState.Deleting;
                         }
                     }
-                    else if (job.State == ClientState.Running || job.State == ClientState.Completed)
+                    else if (job.State == ClientJobState.Running || job.State == ClientJobState.Completed)
                     {
                         var now = DateTime.UtcNow;
 
                         // Clean the job in case the driver is not running
                         if (now - job.LastDriverCommunicationUtc > TimeSpan.FromSeconds(30))
                         {
                             Log($"Driver didn't communicate for {now - job.LastDriverCommunicationUtc}. Halting job.");
-                            job.State = ClientState.Deleting;
+                            job.State = ClientJobState.Deleting;
                         }
                     }
-                    else if (job.State == ClientState.Deleting)
+                    else if (job.State == ClientJobState.Deleting)
                     {
                         Log($"Deleting job {worker?.JobLogText ?? "no worker found"}");
 
                         try
                         {
                             if (worker != null)
                             {
-                                await worker.StopAsync();
+                                // We stop the worker but 
+                                await worker.StopJobAsync();
                             }
                         }
                         finally
                         {
-                            worker?.Dispose();
-                            worker = null;
-
                             _jobs.Remove(job.Id);
                         }
                     }
                 }
                 await Task.Delay(100);
+
+                allJobs = _jobs.GetAll();
+                job = allJobs.FirstOrDefault(clientJob =>
+                {
+                    return string.Equals(clientJob.SpanId, job.SpanId);
```

Comment:
I don't think you will run anymore jobs after the first one finishes.

Reply:
Right, you need to loop for the first available one that matches the current span id, and timeout the current worker if no more job is coming.